File: benchmark-fixtures/go-cli/pkg/logger/logger.go
package logger

import (
	"fmt"
	"os"
	"time"
)

type Logger struct {
	level string
}

func New(level string) *Logger {
	return &Logger{level: level}
}

func (l *Logger) Info(msg string) {
	if l.shouldLog("info") {
		fmt.Printf("[INFO] %s %s\n", time.Now().Format(time.RFC3339), msg)
	}
}

func (l *Logger) Warn(msg string) {
	if l.shouldLog("warn") {
		fmt.Printf("[WARN] %s %s\n", time.Now().Format(time.RFC3339), msg)
	}
}

func (l *Logger) Error(msg string) {
	if l.shouldLog("error") {
		fmt.Fprintf(os.Stderr, "[ERROR] %s %s\n", time.Now().Format(time.RFC3339), msg)
	}
}

func (l *Logger) shouldLog(level string) bool {
	levels := map[string]int{
		"debug": 0,
		"info":  1,
		"warn":  2,
		"error": 3,
	}
	return levels[level] >= levels[l.level]
}

File: benchmark-fixtures/ts-api/src/utils/logger.ts
export class Logger {
  private level: string;

  constructor(level: string = 'info') {
    this.level = level;
  }

  info(message: string): void {
    if (this.shouldLog('info')) {
      console.log(`[INFO] ${new Date().toISOString()} ${message}`);
    }
  }

  warn(message: string): void {
    if (this.shouldLog('warn')) {
      console.warn(`[WARN] ${new Date().toISOString()} ${message}`);
    }
  }

  error(message: string): void {
    if (this.shouldLog('error')) {
      console.error(`[ERROR] ${new Date().toISOString()} ${message}`);
    }
  }

  private shouldLog(level: string): boolean {
    const levels = ['debug', 'info', 'warn', 'error'];
    return levels.indexOf(level) >= levels.indexOf(this.level);
  }
}
